### PR TITLE
chore: Minor cleanup in workflow files.

### PR DESCRIPTION
**Pull request by kinyoklion**

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -32,11 +31,7 @@ jobs:
     steps:
       - uses: actions/checkout@v4
```
Comment (kinyoklion) [inline .github/workflows/publish.yml:32]: Checking things out once is probably sufficient. :-)

File: .github/actions/publish-docs/action.yml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 name: Publish Documentation
-description: 'Publish the documentation to X'
```
Comment (cwaldren-ld) [inline .github/actions/publish-docs/action.yml:2]: Maybe we should start publishing docs to X formerly known as Twitter?

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -32,11 +31,7 @@ jobs:
...
         with:
-          fetch-depth: 0 # If you only need the current version keep this
-
-      - uses: actions/checkout@v4
-        with:
-          fetch-depth: 0 # If you only need the current version keep this
+          fetch-depth: 0
```
Comment (cwaldren-ld) [inline .github/workflows/publish.yml:34]: Curious, does a subsequent step need access to all the tags/history?

**Review by tanderson-ld — APPROVED** [review]

File: .github/workflows/publish.yml (hunk 3 above)
Comment (kinyoklion) [reply .github/workflows/publish.yml:34]: The only thing that needs access to the full fetch depth is actually documentation publishing. It mainly affects monorepos though. If you start like 13 releases at once, like js-core does, then it ends up that changes are being made as it builds the docs, and then has to pull and merge things.

I don't recall the original reason for the fetch-depth addition.

**Pull request merged**